### PR TITLE
KOGITO-3251: Job actions on timeline panel

### PR DESCRIPTION
**Pull request by Sara4994**
Hello,

Please review this PR that adds features to handle jobs actions on timeline panel.

JIRA: https://issues.redhat.com/browse/KOGITO-3251

### CODE REVIEW

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx [hunk 1]
```diff
@@ -85,12 +107,122 @@ const ProcessDetailsTimeline: React.FC<IOwnProps & OUIAProps> = ({
...
+  const handleDetailsToggle = (): void => {
+    setIsDetailsModalOpen(!isDetailsModalOpen);
+  };
+
+  const handleJobReschedule = (job: GraphQL.Job): void => {
+    setSelectedJob(job);
+    handleRescheduleToggle();
+  };
+
+  const handleCancelAction = (job: GraphQL.Job): void => {
```
Comment (nmirasch) [inline ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx:123]: @Sara4994 maybe 'handleJobCancelAction'  to differentiate from node cancelation ?

File: ui-packages/packages/common/src/graphql/types.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable */
+/* tslint:disable */
```
Comment (nmirasch) [inline ui-packages/packages/common/src/graphql/types.tsx]: @Sara4994 I think we moved to eslint

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx (hunk 1 above)
Comment (nmirasch) [reply ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx:123]: @Sara4994 we are duplicating here again, the same calls for the different jobs related actions. Could we unify in one place, at least this calls?

**Review by AjayJagan — APPROVED** [review]
LGTM :+1:

**Comment by Sara4994** [comment]
jenkins please retest this

**Comment by Sara4994** [comment]
jenkins please retest this

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsJobsPanel/ProcessDetailsJobsPanel.tsx [hunk 3]
```diff
@@ -19,12 +19,21 @@ import JobActionsKebab from '../../Atoms/JobActionsKebab/JobActionsKebab';
...
+  Jobs?: GraphQL.Job[];
+}
+interface jobResponse {
+  data: ResponseDataType;
+  loading: boolean;
+  refetch: () => void;
+}
 interface JobsPanelProps {
-  processInstanceId: string;
+  getJobs: jobResponse;
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsJobsPanel/ProcessDetailsJobsPanel.tsx]: the `jobResponse` name is indicating it's a static entity - just response, but it has also `refetch()` function defined. Would a `JobRequest` would be more appropriate?
Also JobsPanelProps property should reflect the name - I would easily name this a `request`. -> getJobs indicate it's rather a function.

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx [hunk 4]
```diff
@@ -29,32 +31,52 @@ import {
...
   >;
+  getJobs: jobResponse;
 }
 enum TitleType {
   SUCCESS = 'success',
   FAILURE = 'failure'
 }
 const ProcessDetailsTimeline: React.FC<IOwnProps & OUIAProps> = ({
   data,
+  getJobs,
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx]: same naming concern as in ProcessDetailsJobsPanel

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx [hunk 5]
```diff
@@ -85,12 +107,109 @@ const ProcessDetailsTimeline: React.FC<IOwnProps & OUIAProps> = ({
...
+    handleRescheduleToggle();
+  };
+
+  const handleCancelAction = (job: GraphQL.Job): void => {
+    jobCancel(job, setCancelModalTitle, setModalContent, getJobs.refetch);
+    handleCancelModalToggle();
+  };
+
+  const handleRescheduleToggle = (): void => {
+    setisRescheduleModalOpen(!isRescheduleModalOpen);
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx]:
```suggestion
    setIsRescheduleModalOpen(!isRescheduleModalOpen);
```

File: ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx [hunk 6]
```diff
@@ -189,15 +310,19 @@ const ProcessDetailsTimeline: React.FC<IOwnProps & OUIAProps> = ({
...
     } else {
-      return [];
+      const items = renderJobActions(node.id, []);
+      return items.flat();
     }
   };
-  const processManagementKebabButtons = (node, index) => {
-    const dropdownItemsValue = dropdownItems(data, node);
+  const processManagementKebabButtons = (node, index): JSX.Element => {
+    const dropdownItemsValue: any = dropdownItems(data, node);
```
Comment (jstastny-cz) [inline ui-packages/packages/management-console/src/components/Organisms/ProcessDetailsTimeline/ProcessDetailsTimeline.tsx]:
```suggestion
    const dropdownItemsValue: JSX.Element[] = dropdownItems(data, node);
```

**Review by jstastny-cz — CHANGES_REQUESTED** [review]
Please check the snapshot ProcessDetailsPage.test.tsx.snap  - the size is not acceptable, caused by passing `getJobs` object into ProcessDetailsJobsPanel component including unnecessary properties IMO (we should just pass the ones needed and defined in interface ProcessDetailsJobsPanel.jobResponse).

And the contents seem suspicious too (e.g. `"error": [Error: Network error: No more mocked responses for the query: query getJobsByProcessInstanceId`).

**Review by nmirasch — APPROVED** [review]

**Review by jstastny-cz — APPROVED** [review]

**Pull request merged**